module/VuFind/src/VuFind/Service/AuthorizationServiceFactory.php

```diff
@@ -0,0 +1,87 @@
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ *
+ * @category VuFind
+ * @package  Service
+ * @author   Juha Luoma <juha.luoma@helsinki.fi>
+ * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
+ * @link     https://vufind.org/wiki/development Wiki
+ */
+
+namespace VuFind\Service;
+
+use Laminas\ServiceManager\Exception\ServiceNotCreatedException;
+use Laminas\ServiceManager\Exception\ServiceNotFoundException;
+use Lmc\Rbac\Mvc\Service\AuthorizationService;
+use Lmc\Rbac\Mvc\Service\AuthorizationServiceFactory as LmcRbacAuthorizationServiceFactory;
+use Psr\Container\ContainerExceptionInterface as ContainerException;
+use Psr\Container\ContainerInterface;
+
+/**
+ * Authorization service factory
+ *
+ * @category VuFind
+ * @package  Service
+ * @author   Demian Katz <demian.katz@villanova.edu>
+ * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
+ * @link     https://vufind.org/wiki/development Wiki
+ */
+class AuthorizationServiceFactory extends LmcRbacAuthorizationServiceFactory
+{
+    /**
+     * Create an object
+     *
+     * @param ContainerInterface $container     Service manager
+     * @param string             $requestedName Service being created
+     * @param null|array         $options       Extra options (optional)
+     *
+     * @return object
+     *
+     * @throws ServiceNotFoundException if unable to resolve the service.
+     * @throws ServiceNotCreatedException if an exception is raised when
+     * creating a service.
+     * @throws ContainerException&\Throwable if any other error occurs
+     */
+    public function __invoke(
+        ContainerInterface $container,
+        $requestedName,
+        ?array $options = null
+    ): AuthorizationService {
+        if (!empty($options)) {
+            throw new \Exception('Unexpected options passed to factory.');
+        }
+        $authorizationService = parent::__invoke($container, $requestedName, $options);
+        $permissions = $container->get(\VuFind\Config\ConfigManagerInterface::class)->getConfigArray('permissions');
+        foreach ($permissions as $settings) {
+            $sectionPermissions = (array)($settings['permission'] ?? []);
+            $assertions = (array)($settings['assertion'] ?? []);
+            if ($sectionPermissions && $assertions) {
+                foreach ($sectionPermissions as $permission) {
+                    foreach ($assertions as $assertion) {
+                        $authorizationService->setAssertion($permission, new $assertion());
```

Comment:
I did a bit of digging, and it looks like deep down, Lmc\Rbac\Service\AuthorizationService contains an AssertionPluginManager. I wonder if we should be registering our assertions in this plugin manager and fetching them from there instead of directly instantiating them. What do you think? Please let me know if you need me to do more digging to figure out exactly _how_ to do this. I imagine there's another factory somewhere that might need adjusting. I can do further research if it would help.

Reply:
seems like it worked, atleast for me!